### PR TITLE
changed postgres driver to pgxpool

### PR DESCRIPTION
**Pull request by mekilis**

### CODE REVIEW

File: database/postgres/postgres.go [hunk 1]
```diff
@@ -15,7 +16,6 @@ import (
 	"github.com/frain-dev/convoy/pkg/log"
 	"github.com/jmoiron/sqlx"
 	_ "github.com/newrelic/go-agent/v3/integrations/nrpq"
```
Comment (jirevwe) [inline database/postgres/postgres.go:18]: I don't think we need this import anymore either

Comment (mekilis) [reply database/postgres/postgres.go:18]: True

File: .gitignore [hunk 2]
```diff
@@ -35,6 +35,6 @@ convoy.json
 docker-compose.yml
 compose.yml
 typesense_data
-postgres_data
+postgres_data**
```
Comment (jirevwe) [inline .gitignore:38]: Won't the below work instead?
```suggestion
postgres_data/
```

**Comment by jirevwe** [comment]
It looks good; we need to test it to see if it performs better.

**Comment by mekilis** [comment]
> It looks good; we need to test it to see if it performs better.

Yea. Still need to fix failing tests

File: api/server_suite_test.go [hunk 3]
```diff
@@ -62,17 +63,23 @@ func getConfig() config.Configuration {
...
+		}
+		_ = os.Setenv("TZ", "") // Use UTC by default :)
 
-	dbHooks := hooks.Init()
-	dbHooks.RegisterHook(datastore.EndpointCreated, func(data interface{}, changelog interface{}) {})
+		dbHooks := hooks.Init()
+		dbHooks.RegisterHook(datastore.EndpointCreated, func(data interface{}, changelog interface{}) {})
 
-	return db
+		pDB = db
```
Comment (danvixent) [inline api/server_suite_test.go:80]: Hmm i'm a bit confused by this func

Comment (mekilis) [reply api/server_suite_test.go:80]: The tests were running out of db connections.

**Review by jirevwe — APPROVED** [review]

**Pull request merged**